apps/shell/tash_main.c

```diff
@@ -151,19 +151,22 @@ char *tash_read_input_line(int fd)
 #if !defined(CONFIG_DISABLE_POLL)
 		FD_ZERO(&tfd);
 		FD_SET(fd, &tfd);
-
+#ifdef CONFIG_PM
+		tash_pm_resume();
```

Comment:
Please add calling pm suspend first.

now code pm resume is first.

Reply:
We cannot use `tash_pm_suspend` here. To enable board sleep, TASH Domain should be resumed here otherwise board will remain in STANDBY state forever.

```
LOOP:
     /* Code which do not give CPU */
     SELECT     // TASH domain should be resumed before this call to enable board sleep
     READ       // TASH domain should be suspended before this call to avoid sleep during read
     /* Code which do not give CPU */

Reply:
I mean, we can add pm suspend during tash init